ballerina/googlesheets_client.bal

```diff
@@ -0,0 +1,520 @@
+                    }
+                } else {
+                    (int|string|decimal|float)|error value =  updateRecord.get(entityKey).ensureType();
+                    if value is error {
+                        return <Error> error(value.message());
+                    }
+                    values.push(value);
+                }
+            }
+            error? response = self.googleSheetClient->updateRowByDataFilter(self.spreadsheetId, self.sheetId, filter, values, "USER_ENTERED");
+            if response is error {
+                return <Error> error(response.message());
+            }
+        } else if (key is map<anydata>) {
+            string metadataValue = self.generateMetadataValue(self.keyFields, key);
+            sheets:DeveloperMetadataLookupFilter filter = {locationType: "ROW", metadataKey: self.tableName, metadataValue: metadataValue};
+            sheets:ValueRange[]|error rows = self.googleSheetClient->getRowByDataFilter(self.spreadsheetId, self.sheetId, filter);
+            if rows is error {
+                return <Error> error(rows.message());
+            }
+            if (rows.length() == 0) {
+                return <Error>error("Error: No element with given key found. ");
+            }
+
+            foreach string entityKey in entityKeys {
+                if (!updateRecord.hasKey(entityKey) && (self.keyFields.indexOf(entityKey) != ())) {
+                    (int|string|decimal)|error value = key.get(entityKey).ensureType();
+                    if value is error {
+                        return <Error> error(value.message());
+                    }
+                    values.push(value);
+                } else if !updateRecord.hasKey(entityKey) && (self.keyFields.indexOf(entityKey) == ()) {
+                    int? indexOfKey = self.fieldMetadata.keys().indexOf(entityKey, 0);
+                    if (indexOfKey !is ()) {
+                        values.push(rows[0].values[indexOfKey]);
+                    }
+                } else {
+                    (int|string|decimal)|error value = updateRecord.get(entityKey).ensureType();
+                    if value is error {
+                        return <Error> error(value.message());
+                    }
+                    values.push(value);
+                }
+            }
+            error? response = self.googleSheetClient->updateRowByDataFilter(self.spreadsheetId, self.sheetId, filter, values, "USER_ENTERED");
+            if response is error {
+                return <Error> error(response.message());
+            }
+
+        }
+    }
+
+    # Performs an SQL `DELETE` operation to delete an entity record from the database.
+    #
+    # + deleteKey - The key used to delete an entity record
+    # + return - `()` if the operation is performed successfully or a `persist:Error` if the operation fails
+    public isolated function runDeleteQuery(anydata deleteKey) returns Error? {
+        if (deleteKey is string|int|decimal) {
```

Comment:
```suggestion
        if deleteKey is string|int|decimal {
```

Reply:
updated in [4346c39](https://github.com/ballerina-platform/module-ballerina-persist/pull/201/commits/4346c3959a465aeb05de0872b94ea1056a715a9a)